cmd/juju/model/configcommand.go

```diff
@@ -301,43 +301,63 @@ func (c *configCommand) setConfig(client configCommandAPI, ctx *cmd.Context) err
 // get writes the value of a single key or the full output for the model to the cmd.Context.
 func (c *configCommand) getConfig(client configCommandAPI, ctx *cmd.Context) error {
 	if len(c.keys) == 1 && certBytes != nil {
-		ctx.Stdout.Write(certBytes)
+		_, _ = ctx.Stdout.Write(certBytes)
 		return nil
 	}
-	attrs, err := client.ModelGetWithMetadata()
+	attrs, err := c.getSkimmedModel(client)
 	if err != nil {
 		return err
 	}
+	attrs, err, finished := c.handleIsKeyOfModel(attrs, ctx)
+	if err != nil {
+		return err
+	} else if attrs != nil && finished {
+		return c.out.Write(ctx, attrs)
+	} else if len(c.keys) > 0 && !finished {
+		if isFileLike(c.keys[0]) {
+			return errors.Errorf("value: %q seems to be a file but not found", c.keys[0])
+		} else {
+			return errors.Errorf("value: %q seems to be neither a file nor a key of the currently targeted model: %q", c.keys[0], attrs["name"])
+		}
+	}
+	return nil
+}
 
+func (c *configCommand) getSkimmedModel(client configCommandAPI) (config.ConfigValues, error) {
+	attrs, err := client.ModelGetWithMetadata()
+	if err != nil {
+		return nil, err
+	}
 	for attrName := range attrs {
-		// We don't want model attributes included, these are available
-		// via show-model.
+		// We don't want model attributes included, these are available via show-model.
 		if c.isModelAttribute(attrName) {
 			delete(attrs, attrName)
 		}
 	}
+	return attrs, nil
+}
+
+func isFileLike(fileLike string) bool {
+	if strings.HasSuffix(fileLike, ".yaml") || strings.HasSuffix(fileLike, ".json") {
```

Comment:
given that this is keys in mongodb I don't know that we support things with a period in them at all. It might be ok just to check for a 3 or 4 char extension as something "file like"

Reply:
You mean something along the line of `strings.HasSuffix(fileLike,".{[a-Z]+}")` instead of checking for yaml and JSON?

Reply:
Yea, maybe even capping that filter at 0-4 chars?

Reply:
done